Sources/SymbolKit/SymbolGraph/SemanticVersion/Prerelease.swift

```diff
@@ -0,0 +1,158 @@
+/*
+ This source file is part of the Swift.org open source project
+ 
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ 
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+extension SymbolGraph.SemanticVersion {
+    /// A storage for pre-release identifiers.
+    internal struct Prerelease {
+        /// The identifiers.
+        internal let identifiers: [Identifier]
+        /// A pre-release identifier.
+        internal enum Identifier {
+            /// A numeric pre-release identifier.
+            /// - Parameter identifier: The identifier.
+            case numeric(_ identifier: UInt)
+            /// An alphanumeric pre-release identifier.
+            /// - Parameter identifier: The identifier.
+            case alphanumeric(_ identifier: String)
+        }
+    }
+}
+
+// MARK: - Initializers
+
+extension SymbolGraph.SemanticVersion.Prerelease {
+    /// Creates a semantic version pre-release from the given pre-release string.
+    /// - Note: Empty string translates to an empty pre-release identifier, which is invalid.
+    /// - Parameter dotSeparatedIdentifiers: The given pre-release string to create a semantic version pre-release from.
+    /// - Throws: A `SymbolGraph.SemanticVersionError` instance if `dotSeparatedIdentifiers` is not a valid pre-release string.
+    internal init(_ dotSeparatedIdentifiers: String?) throws {
+        guard let dotSeparatedIdentifiers = dotSeparatedIdentifiers else {
```

Comment:
I don't think the internal representation of the pre-release information has much to do with this initializer signature. Rather, the reason that `dotSeparatedIdentifiers` is optional is because the pre-release string it's an optional information in the SymbolKit's OpenAPI documentation, and decoded as a `String?` by SymbolKit. The only way to make it non-optional imo is moving the optional checking outside to right before the call site.

Reply:
See [this comment](https://github.com/apple/swift-docc-symbolkit/pull/36#discussion_r889710443): if the  prerelease information on semantic was optional array of string then this initializer wouldn't need to check for nil values. Instead, if the prerelease string was `nil` then no Prerelease would be initialized. This could for example be done with optional map at the call site.